frontend/src/service/Url.service.ts

```diff
@@ -179,15 +185,15 @@ export class UrlService {
   private gqlCreateURLVariable(
     captchaResponse: string,
     link: Url,
-    alias: string | null,
     isPublic: boolean = false
   ) {
+    let linkAlias = link.alias === '' ? null : link.alias!;
     return {
       captchaResponse: captchaResponse,
       authToken: this.authService.getAuthToken(),
       urlInput: {
         originalURL: link.originalUrl,
-        customAlias: alias
+        customAlias: linkAlias
```

Comment:
```suggestion
        customAlias: alias
```

Reply:
I wonder why can't we directly pass alias?

Reply:
This is what happens when you specify an empty custom alias and pass it directly into `urlInput` argument:

![image](https://user-images.githubusercontent.com/3276350/80825439-66978d80-8bae-11ea-8413-2cc387898240.png)

Looks something like this in `url` table in DB:

![image](https://user-images.githubusercontent.com/3276350/80825473-78793080-8bae-11ea-8714-f68980e53d48.png)

If we instead pass `null` as the argument, this behaviour will be prevented and an unknown error will be thrown instead:

![image](https://user-images.githubusercontent.com/3276350/80825580-afe7dd00-8bae-11ea-8fac-d926d9c54bf4.png)

Should we instead throw an explicit error when user passes in an empty custom alias? That would probably be the best way to deal with this case I think.

Reply:
@Coteh I think we need to fix the backend alias validation.

Reply:
Let's open another PR to fix the backend!

Reply:
I see. This code exists before your changes.

Let's add a TODO to fix the backend validation code. We can't always trust the input from the frontend because malicious users can use JavaScript to modify our page: https://martinfowler.com/articles/web-security-basics.html. Please read `Trust` and `Input Validation` part. At the backend, we should treat empty alias as automatic alias.

Reply:
Issue created https://github.com/short-d/short/issues/728